### ISSUE TITLE
SIGSEGV error when running tutorial

### ISSUE THREAD

**Issue by AfairiJJ** [sdv-dev/SDV#1287]
### Environment Details

Please indicate the following details about the environment in which you found the bug:
 
* SDV version: 0.18.0
* Python version: Python 3.10.9
* Operating System: mac OS Ventura 13.2 (22D49)
* Hardware: MacBook Air M2 (2022) 

### Error Description
When trying to run the usage example part of the [tutorial](https://github.com/sdv-dev/CTGAN) on my mac: 
I get SIGSEGV (small input dataset) and SIGABRT (large input dataset) errors early on in the process (after around 10 seconds). 

### Steps to reproduce

I followed the Github SDV [tutorial](https://github.com/sdv-dev/CTGAN) and got stuck at the usage example.

In:
```
from ctgan import CTGAN
from ctgan import load_demo
import faulthandler
real_data = load_demo().head(20)

# Names of the columns that are discrete
discrete_columns = [
    'workclass',
    'education',
    'marital-status',
    'occupation',
    'relationship',
    'race',
    'sex',
    'native-country',
    'income'
]

ctgan = CTGAN(epochs=10)
faulthandler.enable()
ctgan.fit(real_data, discrete_columns)

# Create synthetic data
synthetic_data = ctgan.sample(1000)
```

Out:
```
Fatal Python error: Segmentation fault

Thread 0x000000020571b2c0 (most recent call first):
  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/sklearn/cluster/_kmeans.py", line 731 in _kmeans_single_lloyd
  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/sklearn/cluster/_kmeans.py", line 1468 in fitFatal Python error: Segmentation fault


  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/sklearn/mixture/_base.py", line 119 in _initialize_parameters
  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/sklearn/mixture/_base.py", line 241 in 
Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)
```

In:
```
from ctgan import CTGAN
from ctgan import load_demo
import faulthandler
real_data = load_demo() #.head(20)

# Names of the columns that are discrete
discrete_columns = [
    'workclass',
    'education',
    'marital-status',
    'occupation',
    'relationship',
    'race',
    'sex',
    'native-country',
    'income'
]

ctgan = CTGAN(epochs=10)
faulthandler.enable()
ctgan.fit(real_data, discrete_columns)

# Create synthetic data
synthetic_data = ctgan.sample(1000)
```

Out:
```
Fatal Python error: Aborted

Fatal Python error: Thread 0xAborted

0000000200c342c0 (most recent call first):
  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/sklearn/cluster/_kmeans.py", line 681 in _kmeans_single_lloyd
  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/sklearn/cluster/_kmeans.py", line 1468 in fit
  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/sklearn/mixture/_base.py", line 119 in _initialize_parameters
  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/sklearn/mixture/_base.py", line 241 in fit_predict
  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/sklearn/mixture/_base.pyFatal Python error: Aborted

", line 186 in fit
  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/rdt/transformers/numerical.py", line 442 in _fit
  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/rdt/transformers/base.py", line 333 in fit
  File "/Users/janjaniszewski/opt/anaconda3/envs/Thesis/lib/python3.10/site-packages/rdt/transformers/base.py"
Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```

**Comment by npatki** [comment]
Hi @AfairiJJ thanks for the detailed traces. The `SIGSEGV` doesn't seem to be related to memory. A few members of the SDV team have the same setup as you and we weren't able to replicate it.

I'm wondering if there's anything else installed in your environment that may be causing issues. Would you be able to create a new, clean environment and just install the SDV?

**Comment by npatki** [comment]
Hi @AfairiJJ, we still haven't been able to replicate the issue but we have released a new, [**SDV 1.0**](https://docs.sdv.dev/sdv) with significant changes to the way we process and store data. I'd recommend upgrading to this library and re-running your workflow.

I'll close the issue as stale but if you continue to run into `SIGSEGV` in the new SDV 1.0 version, please reply and we'll reopen the issue for investigation.
